### ISSUE TITLE
cartesian config parse error - filters process is broken

### ISSUE THREAD

**Issue by Andrei-Stepanov** [avocado-framework/avocado-vt#134]
Hello.
Bug is incredible.
It breaks basic principles and rules.

Config example:

```
# cat 1.cfg 
variants:
    - one:
    - two:

variants:
    - x:
    - y:
    - z:

only one..y, two
only one
```

Invoke:

```
# ./virttest/cartesian_config.py 1.cfg 
NOTHING
```

You can play with the example.

It works fine for:
 * **only one..x, two**
 * remove second **only one**
 * **one..y, one..x, two**

It fails for:
```
only one..y, two
only one
```
OR
```
only one..z, two
only one
```

**Comment by Andrei-Stepanov** [comment]
I have found commit that had broken right behavior:
```
git log --oneline virttest/cartesian_config.py                                                                                   
1050105 Fixes for join, suffix operators
26aceb1 Add suffix & join to cartesian_config.py
cc30cbb virttest: the params of docstring should begin with colon
6ccba7e cartesian_config: Pylint W0142 no longer need disable
389ee94 PEP8 fixes
4e09e7a virt: Support _min _max _fixed suffix in configure.
cc0ae21 virttest: Mass fix PEP8 E128 errors
427a782 virttest: Mass PEP8 fix for E713
529ab89 virttest: PEP8 Mass fix - part 1
3c1c2c2 virttest: Docstring fixes
3feece7 Mass PEP8 fixes
3b40302 virttest.cartesian_config: Make 'replace deprecated template with eval in cart-config' works for multi deprecated template
0f43942 virt: Makes ./run list-test and others list test function less verbose.
5af8a1e virt: Fix test directory tag name.
0a355b4 virt: Adds cart config variant name to file mapping.
301804e More docstring fixes
811faf7 Mass docstring fix
6b89dcf PEP8 mass fixes
a02628f Mass spelling fixes
64e93f8 virt: Cart config memory and code style optimization.
64d2e7a virttest.cartesian_config: Remove unnecessary pylint error supression
6154fe8 virttest.cartesian_config: Supress invalid pylint option on python 2.4
2a17857 virt: Fix bug in cart config.
9c24f76 virt: Replace deprecated template with eval in cart-config.
43693a3 virt.cartesian_config: Fix a typo in this module
7d89f25 virt: New version Cartesian config.
c262431 virt: Cartesian config speed and memory optimization.
0199013 virt: Cartesian config bug repair.
d9f4375 cartesian_config.py: add -r option for "repr mode"
702fa7c cartesian_config: Keep track of extra config parsed programatically
65fd075 virt: cartesian_config catch another cfg error.
3ef2b2f cartesian_config: Fix use of reserved keywords and unused variables
580de09 virt: cartesian_config: Record filename on cartesian objects
2d46161 virt: Moving cartesian parser to virttest library
```


```
# git show d9f4375:virttest/cartesian_config.py > d9f4375.py                                                            
# git show 0199013:virttest/cartesian_config.py > 0199013.py
# chmod 755 d9f4375.py 0199013.py
# ./d9f4375.py 1.cfg 
dict    1:  y.one
# ./0199013.py 1.cfg
Nothing
```

Bug was introduced 2 years ago, by commit : 0199013    

https://github.com/avocado-framework/avocado-vt/commit/0199013

**Comment by Andrei-Stepanov** [comment]
I believe commit : 0199013 is NOT the source of the problem.
The bug revealed with commit : 0199013, but it is not source of it.
The problem existed before.

We should remove whole elif block at https://github.com/avocado-framework/avocado-vt/blob/master/virttest/cartesian_config.py#L2160

I have sent PR: https://github.com/avocado-framework/avocado-vt/pull/137

**Comment by lmr** [comment]
Fixed by merging of #137
